SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityScheduler.java

```diff
@@ -569,19 +584,33 @@ private boolean shouldRetryImmediately(SingularityRequest request, SingularityDe
     return true;
   }
 
-  private int getNumMissingInstances(List<SingularityTaskId> matchingTaskIds, SingularityRequest request, SingularityPendingRequest pendingRequest) {
+  private int getNumMissingInstances(List<SingularityTaskId> matchingTaskIds, SingularityRequest request, SingularityPendingRequest pendingRequest, Optional<SingularityPendingDeploy> maybePendingDeploy) {
     if (request.isOneOff() && pendingRequest.getPendingType() == PendingType.ONEOFF) {
       return 1;
     }
 
-    final int numInstances = request.getInstancesSafe();
+    final int numInstances;
+    if (maybePendingDeploy.isPresent() && !(maybePendingDeploy.get().getCurrentDeployState() == DeployState.CANCELED) && maybePendingDeploy.get().getDeployProgress().isPresent()) {
```

Comment:
a good indicator of a place where you can make a method is when you have:
final XYZ object;

and then a bunch of if code - turn that if code into the method